test/remote/gateways/remote_hi_pay_test.rb

```diff
@@ -0,0 +1,112 @@
+require 'test_helper'
+
+class RemoteHiPayTest < Test::Unit::TestCase
+  def setup
+    @gateway = HiPayGateway.new(fixtures(:hi_pay))
+    @bad_gateway = HiPayGateway.new(username: 'bad', password: 'password')
+
+    @amount = 500
+    @credit_card = credit_card('4111111111111111', verification_value: '514', first_name: 'John', last_name: 'Smith', month: 12, year: 2025)
+    @bad_credit_card = credit_card('5144144373781246')
+
+    @options = {
+      order_id: "Sp_ORDER_#{SecureRandom.random_number(1000000000)}",
+      description: 'An authorize',
+      email: 'john.smith@test.com'
+    }
+
+    @billing_address = {
+      company: 'testCompany',
+         address1: '1 Main St.',
+         city: 'Burlington',
+         state: 'MA',
+         country: 'US',
+         zip: '01803-3747',
+         phone: '1234567890'
+    }
```

Comment:
There's an `address` helper method so you don't have to build this out each time. Also, if the gateway is expecting data in a certain format, there should be formatter methods to also help, like with the `phone` number.

Reply:
Oh, I'll use it then.